x/emissions/keeper/inference_synthesis/network_inferences.go

```diff
@@ -1,200 +1,243 @@
 		},
-		Reputer:                       "allo1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqas6usy",
-		ExtraData:                     nil,
-		CombinedValue:                 alloraMath.ZeroDec(),
-		InfererValues:                 make([]*emissions.WorkerAttributedValue, 0),
-		ForecasterValues:              make([]*emissions.WorkerAttributedValue, 0),
+		Reputer:       "allo1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqas6usy",
+		CombinedValue: medianValue,
+		InfererValues: fn.Map(inferences.Inferences, func(inf *emissions.Inference) *emissions.WorkerAttributedValue {
+			return &emissions.WorkerAttributedValue{Worker: inf.Inferer, Value: inf.Value}
+		}),
+		ForecasterValues:              make([]*emissions.WorkerAttributedValue, 0), // TODO(spook): can all of these be nil?
 		NaiveValue:                    alloraMath.ZeroDec(),
-		OneOutInfererValues:           make([]*emissions.WithheldWorkerAttributedValue, 0),
-		OneOutForecasterValues:        make([]*emissions.WithheldWorkerAttributedValue, 0),
-		OneInForecasterValues:         make([]*emissions.WorkerAttributedValue, 0),
-		OneOutInfererForecasterValues: make([]*emissions.OneOutInfererForecasterValues, 0),
+		OneOutInfererValues:           make([]*emissions.WithheldWorkerAttributedValue, 0), // TODO(spook): can all of these be nil?
+		OneOutForecasterValues:        make([]*emissions.WithheldWorkerAttributedValue, 0), // TODO(spook): can all of these be nil?
+		OneInForecasterValues:         make([]*emissions.WorkerAttributedValue, 0),         // TODO(spook): can all of these be nil?
+		OneOutInfererForecasterValues: make([]*emissions.OneOutInfererForecasterValues, 0), // TODO(spook): can all of these be nil?
+		ExtraData:                     nil,
 	}
+	return &GetNetworkInferencesResult{
+		NetworkInferences:                    networkInferences,
+		ForecasterToForecastImpliedInference: nil,
+		InfererToWeight:                      nil,
+		ForecasterToWeight:                   nil,
+		InferenceBlockHeight:                 inferenceBlockHeight,
+		LossBlockHeight:                      0,
+	}, nil
+}
 
-	forecasterToForecastImpliedInference = make(map[string]*emissions.Inference, 0)
-
-	// Add inferences to the bundle
-	for _, inference := range inferences.Inferences {
-		networkInferences.InfererValues = append(networkInferences.InfererValues, &emissions.WorkerAttributedValue{
-			Worker: inference.Inferer,
-			Value:  inference.Value,
-		})
-	}
+func calcNetworkInferencesMultiple(
+	ctx sdk.Context,
+	k emissionskeeper.Keeper,
+	topicId TopicId,
+	inferences *emissions.Inferences,
+	inferenceBlockHeight BlockHeight,
+	networkLosses *emissions.ValueBundle,
+) (*GetNetworkInferencesResult, error) {
+	// TODO(spook): should the following fetches happen at the very top of the
+	// call stack to ensure that the topic is real and the module params are present?
+	// It would waste some i/o for the other cases (single, multiple+median), but
+	// would be more correct and could surface issues more readily.
```

Comment:
Yes it's cleaner but there are lots of divergences from clean practice in heavily compute-constrained environments, I lean on leaving as-is so scope is filled with more of "what's needed, when it's needed"